### PR TITLE
Adding an example for a JSON function handler

### PR DESCRIPTION
**Pull request by subfuzion**

### CODE REVIEW

**Comment by kevmoo** [comment]
I don't know why GitHub has `assign` and `reviewers` weird

Is this `Ready for review`?

**Comment by kevmoo** [comment]
@subfuzion – mark as ready for review?

**Comment by subfuzion** [comment]
Reviewers are for requesting someone to review; Assigness should generally be self-assigned when you accept the request.
I'll mark ready for review when ready, which it is now. :D I've updated the readme, but will defer updating the docs for another PR.

**Comment by subfuzion** [comment]
Here's a [link](https://github.com/GoogleCloudPlatform/functions-framework-dart/tree/209585b9f7e366d8b6f21ce5ca5c9684d6f96ff1/examples/json) (for convenience) to the formatted docs for the latest commit.

File: examples/json/README.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+Finally, the function creates an object returns it. The Functions Framework will
+take this and attempt "do the right" thing. In this case, the function is typed
+to return a `FutureOr<GreetingResponse>`, which has a `toJson()` method, so the
+framework will invoke this, then set the response body to the stringified result
+and set the response header (`content-type`) to `application/json`).
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
```
Comment (kevmoo) [inline examples/json/README.md:54]: Since this is not async, you can just return `GreetingResponse`

File: examples/json/test/data/response.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "salutation": "Hello",
```
Comment (kevmoo) [inline examples/json/test/data/response.json:2]: don't need this file, right?

File: examples/json/test/data/request.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "name": "World"
```
Comment (kevmoo) [inline examples/json/test/data/request.json:2]: Don't need this file, right?

**Review by kevmoo — CHANGES_REQUESTED** [review]

File: examples/json/README.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  final name = request['name'] as String ?? 'World';
+  final json = GreetingResponse(salutation: 'Hello', name: name);
+  return json;
+}
+```
+
+The full code is shown below:
+
+_lib/functions.dart_
```
Comment (kevmoo) [inline examples/json/README.md:63]: the markdown linter doesn't like this 🤷

Comment (subfuzion) [reply examples/json/README.md:63]: The linter is wrong! I don't like how it converted the filename to a Dart comment for `example/hello` 😖

Comment (kevmoo) [reply examples/json/README.md:63]: 🤷

File: examples/json/README.md (hunk 1 above)
Comment (subfuzion) [reply examples/json/README.md:54]: True and yet 99% of the time there's going to be I/O and you're going to want a future. But you're right; when we have an example with some I/O, we can show that properly. Wondering if I should call it out here though... 🤔 Whaddaya think?

File: examples/json/test/data/response.json (hunk 2 above)
Comment (subfuzion) [reply examples/json/test/data/response.json:2]: good catch (for both), meant to delete. Thanks!

File: examples/json/README.md (hunk 1 above)
Comment (subfuzion) [reply examples/json/README.md]: Grateful to the linter for finding my link mistakes though! Hang on a moment.

File: examples/json/README.md (hunk 4 above)
Comment (subfuzion) [reply examples/json/README.md]: Grateful to the linter for finding my link mistakes at least! Hang on a moment.

File: examples/json/README.md (hunk 1 above)
Comment (subfuzion) [reply examples/json/README.md]: Alrighty then, fixed.

**Review by kevmoo — APPROVED** [review]

**Comment by kevmoo** [comment]
:shipit:

**Comment by kevmoo** [comment]
@subfuzion – I did a force push to cleanup the history here. FYI if/when your sync is weird...

**Pull request merged**